docs/components/security.rst

```diff
@@ -1,23 +1,365 @@
+                       'full'          => 'mautic.core.permissions.full',
+                   ),
+                   'label'   => 'plugin.helloworld.permissions',
+
+                   // Set existing data
+                   'data'    => (!empty($data['worlds']) ? $data['worlds'] : array()),
 
-``mautic.permissions`` tag
-Class must extend ``\Mautic\CoreBundle\Security\Permissions\AbstractPermissions``
+                   // Bundle name (used to build frontend form)
+                   'bundle'  => 'helloWorld',
 
+                   // Permission level (used to build frontend form)
+                   'level'   => 'worlds'
+               )
+           );
+       }
+
+       /**
+        * Permission set identifier; should be bundleName
+        * 
+        * @return string
+        */
+       public function getName()
+       {
+           return 'helloWorld';
+       }
+   }
+
+You can register the permission class by adding it to your ``config.php`` as shown below. Make sure it's in the ``services.permissions`` group so that Mautic can pick it up correctly.
+
+.. code-block:: php
+
+    <?php
+    // plugins/HelloWorldBundle/Config/config.php
+
+    return [
+        ...
+        'services' => [
+            'permissions' => [
+                'marketplace.permissions' => [
+                    'class' => \MauticPlugin\HelloWorldBundle\Security\Permissions\WorldsPermissions::class,
+                ],
+            ],
+        ],
+    ];
+
+You can learn more about the available options by looking at the ``Mautic\CoreBundle\Security\Permissions\AbstractPermissions`` PHPDoc, but here's the most important information:
+
+**__construct()**
+
+The construct method should do two things. It should call ``parent::__construct($params)`` and/or it should set ``$this->params = $params;``. 
+
+Then it should define ``$this->permissions``. ``$this->permissions`` is an array of permission levels that are each arrays with permissions assigned to bits. 
+For example, in the code block, a custom permission level of ``worlds`` gets defined with the permissions of ``use_telescope``, ``send_probe``, ``visit`` and ``full``.
+To validate whether a User has permission to the level ``worlds`` and permission ``send_probe`` , ``$mauticSecurity->isGranted('plugin:helloWorld:worlds:send_probe')`` would be used.
+
+Mautic provides a few helper methods for common permission sets:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Method
+     - Description
+   * - ``addStandardPermissions()``
+     - Set view, edit, create, delete, publish - with option to exclude, and full permissions.
+   * - ``addExtendedPermissions()``
+     - Set creator level restrictions: ``viewown``, ``viewother``, ``editown``, ``editother``, ``create``, ``deleteown``, ``deleteother``, ``publishown`` - with option to exclude, ``publishother`` - with option to exclude, and full
+   * - ``addManagePermission()``
+     - Set single manage permission
```

Comment:
Not sure what this means, it seems incomplete unless 'single manage permission' is a thing.

Reply:
Good catch - just looked at the code and it's basically an "all or nothing" approach, so I went with the following:

> Add a single ``manage`` permission, which is the same as ``full``. Use this in cases where you only need a single permission for everything, also known as an "all or nothing" approach.

Reply:
Great, thanks!